packages/plugin-ext/src/plugin/terminal-ext.ts

```diff
@@ -16,38 +16,59 @@
 import { UUID } from '@theia/core/shared/@phosphor/coreutils';
 import { inject, injectable } from '@theia/core/shared/inversify';
 import { Terminal, TerminalOptions, PseudoTerminalOptions, ExtensionTerminalOptions, TerminalState } from '@theia/plugin';
-import { TerminalServiceExt, TerminalServiceMain, PLUGIN_RPC_CONTEXT } from '../common/plugin-api-rpc';
+import { TerminalServiceExt, TerminalServiceMain, PLUGIN_RPC_CONTEXT, Plugin } from '../common/plugin-api-rpc';
 import { RPCProtocol } from '../common/rpc-protocol';
 import { Event, Emitter } from '@theia/core/lib/common/event';
 import { MultiKeyMap } from '@theia/core/lib/common/collections';
 import { Deferred } from '@theia/core/lib/common/promise-util';
 import * as theia from '@theia/plugin';
 import * as Converter from './type-converters';
-import { Disposable, EnvironmentVariableMutatorType, TerminalExitReason, ThemeIcon } from './types-impl';
+import { Disposable, EnvironmentVariableMutatorType, TerminalExitReason, ThemeIcon, URI } from './types-impl';
 import { NO_ROOT_URI, SerializableEnvironmentVariableCollection } from '@theia/terminal/lib/common/shell-terminal-protocol';
 import { ProvidedTerminalLink } from '../common/plugin-api-rpc-model';
-import { ThemeIcon as MonacoThemeIcon } from '@theia/monaco-editor-core/esm/vs/base/common/themables';
+import { ThemeIcon as MonacoThemeIcon, ThemeColor as MonacoThemeColor } from '@theia/monaco-editor-core/esm/vs/base/common/themables';
+import { PluginSharedStyle } from '../main/browser/plugin-shared-style';
+import { PluginIconPath } from './plugin-icon-path';
+import { IconUrl } from '../common';
+import { DisposableCollection } from '@theia/core';
 
-export function getIconUris(iconPath: theia.TerminalOptions['iconPath']): { id: string } | undefined {
+export function getIconUris(iconPath: theia.TerminalOptions['iconPath']): MonacoThemeIcon | IconUrl | undefined {
     if (ThemeIcon.is(iconPath)) {
         return { id: iconPath.id };
+    } else if (typeof iconPath === 'object' && 'light' in iconPath) {
+        return { light: iconPath.light.toString(), dark: iconPath.dark.toString() };
+    } else if (typeof iconPath === 'string') {
+        return iconPath;
     }
     return undefined;
 }
 
-export function getIconClass(options: theia.TerminalOptions | theia.ExtensionTerminalOptions): string | undefined {
-    const iconClass = getIconUris(options.iconPath);
-    if (iconClass) {
-        return MonacoThemeIcon.asClassName(iconClass);
+export function getIconClass(
```

Comment:
The definition of the icon in the TerminalOptions is basically the same as in the QuickInputItem. Can't we reuse the same logic for transferring and and resolving icons here? It would make sense (IMO) to solve this the same way everywhere it applies.

Reply:
Doing stuff in a separate PR is fine with me, as long as we have a clear timeline on when it's going to happen. I wouldn't just want to "throw it on the pile".

Reply:
We now support the quick pick `iconPath` property.